mis/solver/solver.py

```diff
@@ -0,0 +1,223 @@
+            types=cast(  # Type annotation in cplex is wrong
+                str, [c.variables.type.binary] * len(solved_graph.nodes())
+            ),
+        )
+
+        # Independence constraints
+        c.linear_constraints.add(
+            lin_expr=[
+                cplex.SparsePair(ind=[str(u), str(v)], val=[1.0, 1.0])
+                for u, v in solved_graph.edges()
+            ],
+            senses=["L"] * solved_graph.number_of_edges(),
+            rhs=[1.0] * solved_graph.number_of_edges(),
+        )
+
+        # Objective function definition
+        c.objective.set_linear([(str(node), 1) for node in solved_graph.nodes()])
+        c.objective.set_sense(c.objective.sense.maximize)
+
+        # Solve MIP without logs
+        c.set_log_stream(None)
+        c.set_results_stream(None)
+        c.set_warning_stream(None)
+        c.solve()
+
+        # Extract solution
+        solution_values = c.solution.get_values()
+        selected_nodes = [node for node, value in enumerate(solution_values) if value >= 0.9]
+
+        # Convert back to original node labels
+        conversion_table = list(graph.nodes())
+        return Execution.success(
+            [
+                MISSolution(
+                    original=graph,
+                    nodes=[conversion_table[node] for node in selected_nodes],
+                    energy=0,
+                )
+            ]
+        )
+
+
+class MISSolverQuantum(BaseSolver):
+    """
+    Quantum solver that orchestrates the solving of a MISproblem using
+    embedding, pulse shaping, and quantum execution pipelines.
+    """
+
+    def __init__(self, instance: MISInstance, config: SolverConfig):
+        """
+        Initialize the MISSolver with the given problem and configuration.
+
+        Args:
+            instance (MISInstance): The MISproblem to solve.
+            config (SolverConfig): Solver settings including backend and
+                device.
+        """
+        assert config.backend is not None
+        if config.embedder is None:  # FIXME: That's a side-effect on config
+            config.embedder = DefaultEmbedder()
+        if config.pulse_shaper is None:
+            config.pulse_shaper = DefaultPulseShaper()
+        if config.device is None:
+            config.device = config.backend.device()
+
+        super().__init__(instance, config)
+
+        self.fixtures = Fixtures(instance, self.config)
+        self._register: Register | None = None
+        self._pulse: Pulse | None = None
+        self._solution: MISSolution | None = None
+
+        # FIXME: Normalize embedder.
+
+    def embedding(self) -> Register:
+        """
+        Generate a physical embedding (register) for the MISvariables.
+
+        Returns:
+            Register: Atom layout suitable for quantum hardware.
+        """
+        # FIXME: mypy seems to have an issue here, need to investigate.
+        config: SolverConfig = self.config
+        embedder = config.embedder
```

Comment:
```suggestion
        embedder = self.config.embedder
```

Reply:
Yeah, I've had all sorts of issues with mypy not figuring out the type of anything in this method, I need to investigate.